crates/c-api/src/val.rs

```diff
@@ -30,48 +60,46 @@ impl Default for wasm_val_t {
 }
 
 impl wasm_val_t {
-    pub fn from_val(val: &Val) -> wasm_val_t {
+    pub fn from_val(val: Val) -> wasm_val_t {
         match val {
             Val::I32(i) => wasm_val_t {
                 kind: from_valtype(&ValType::I32),
-                of: wasm_val_union { i32: *i },
+                of: wasm_val_union { i32: i },
             },
             Val::I64(i) => wasm_val_t {
                 kind: from_valtype(&ValType::I64),
-                of: wasm_val_union { i64: *i },
+                of: wasm_val_union { i64: i },
             },
             Val::F32(f) => wasm_val_t {
                 kind: from_valtype(&ValType::F32),
-                of: wasm_val_union { u32: *f },
+                of: wasm_val_union { u32: f },
             },
             Val::F64(f) => wasm_val_t {
                 kind: from_valtype(&ValType::F64),
-                of: wasm_val_union { u64: *f },
+                of: wasm_val_union { u64: f },
+            },
+            Val::ExternRef(r) => wasm_val_t {
+                kind: from_valtype(&ValType::ExternRef),
+                of: wasm_val_union {
+                    ref_: Box::into_raw(Box::new(wasm_ref_t {
+                        r: WasmRefInner::ExternRef(r),
+                    })),
+                },
+            },
+            Val::FuncRef(f) => wasm_val_t {
+                kind: from_valtype(&ValType::FuncRef),
+                of: wasm_val_union {
+                    ref_: Box::into_raw(Box::new(wasm_ref_t {
+                        r: WasmRefInner::FuncRef(f),
+                    })),
+                },
             },
             _ => unimplemented!("wasm_val_t::from_val {:?}", val),
         }
     }
 
     pub fn set(&mut self, val: Val) {
```

Comment:
Is this method still used in the API somewhere? This seems like it's probably unsafe to do since it runs the dtor on the previous contents and I don't think that there's anywhere else we should do that in the API